csrc/preseg_passes/make_resharding_contiguous.cpp

```diff
@@ -109,6 +109,7 @@ void setLoopAndAllocationDomain(TensorView* tv) {
       " as ",
       alloc_dom);
   tv->reorder(permutation.value());
+  reorderDIDToFront(tv);
```

Comment:
This would not work.

We have a limitation that loop == allocation in segmentation (and potentially downstream).
When DID is reordered to the front, that is applicable to both loop and allocation. This will break `getShardedLogicalAxis` for logical DID split. I'll have to think more about how to do this here.

Reply:
I'm reading through the errors. Many errors are caused contiguity not being reordered the same way.